column_buffer.go

```diff
@@ -590,10 +590,10 @@ func (col *repeatedColumnBuffer) Less(i, j int) bool {
 	row2Length := repeatedRowLength(col.repetitionLevels[row2.offset:])
 
 	for k := 0; k < row1Length && k < row2Length; k++ {
-		x := int(row1.offset) + k
-		y := int(row2.offset) + k
-		definitionLevel1 := col.definitionLevels[j+k]
-		definitionLevel2 := col.definitionLevels[j+k]
+		x := int(row1.baseOffset)
+		y := int(row2.baseOffset)
+		definitionLevel1 := col.definitionLevels[int(row1.offset)+k]
+		definitionLevel2 := col.definitionLevels[int(row2.offset)+k]
```

Comment:
The wrong offsets were being used: `.offset` is the position within the definition levels, `.baseOffset` is the position of the row values in the underlying column buffer.